From 7b6d728f257a92fb2ac8cc916e70117019c9de54 Mon Sep 17 00:00:00 2001
From: David S. Wang <dsw@cloudera.com>
Date: Wed, 27 Mar 2013 09:04:01 -0700
Subject: [PATCH 13/96] HBASE-5879 Enable JMX metrics collection for the Thrift proxy

Reason: Improvement
Author: Mikhail Bautin
Ref: CDH-8723
(cherry picked from commit 6312caa02303a0d8bed1f0b0f2399425bd3ba21f)
---
 bin/hbase-config.sh |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/bin/hbase-config.sh b/bin/hbase-config.sh
index c5cbbc4..149d515 100644
--- a/bin/hbase-config.sh
+++ b/bin/hbase-config.sh
@@ -75,6 +75,14 @@ HBASE_CONF_DIR="${HBASE_CONF_DIR:-$HBASE_HOME/conf}"
 HBASE_REGIONSERVERS="${HBASE_REGIONSERVERS:-$HBASE_CONF_DIR/regionservers}"
 # List of hbase secondary masters.
 HBASE_BACKUP_MASTERS="${HBASE_BACKUP_MASTERS:-$HBASE_CONF_DIR/backup-masters}"
+# Thrift jmx opts
+if [ -z "$HBASE_THRIFT_JMX_OPTS" ]; then
+  HBASE_THRIFT_JMX_OPTS="$HBASE_JMX_OPTS -Dcom.sun.management.jmxremote.port=8093"
+fi
+# Thrift opts
+if [ -z "$HBASE_THRIFT_OPTS" ]; then
+  export HBASE_THRIFT_OPTS="$HBASE_THRIFT_JMX_OPTS"
+fi
 
 # Source the hbase-env.sh.  Will have JAVA_HOME defined.
 if [ -f "${HBASE_CONF_DIR}/hbase-env.sh" ]; then
-- 
1.7.0.4

